client/components/data/query-whois/index.jsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingWhois } from 'state/selectors';
+import { requestWhois } from 'state/domains/management/actions';
+
+class QueryWhois extends Component {
+	componentWillMount() {
+		if ( this.props.requesting ) {
+			return;
+		}
+		this.props.requestWhois( this.props.domain );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+QueryWhois.propTypes = {
+	domain: PropTypes.string,
+	requesting: PropTypes.bool,
+	requestWhois: PropTypes.func
+};
+
+export default connect(
+	( state, { domain } ) => ( { requesting: isRequestingWhois( state, domain ) } ),
```

Comment:
Oh, I'm so sorry, @hewsut and @deBhal. I actually completely misread that line. I'll blame it on the poor reading experience on my phone… maybe, or just my stupid brain. I thought I'd read some function being created in there. That `connect` call is actually totally fine.

That must've been some really confusing feedback. Apologies. 😓